common/libradoc/tests/update.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_documentation_tool as libradoc;
+use serde::Deserialize;
+use serde_generate::rust;
+use serde_reflection::{Samples, Tracer, TracerConfig};
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+enum MyEnum {
+    Unit,
+    Newtype(MyStruct),
+    Tuple(u16, Option<bool>),
+    Struct { a: u32 },
+    NewTupleArray((u16, u16, u16)),
+}
+
+#[derive(Deserialize)]
+struct MyStruct(u64);
+
+#[test]
+fn test_doctool() {
+    let mut tracer = Tracer::new(TracerConfig::default());
+    let samples = Samples::new();
+    tracer.trace_type::<MyEnum>(&samples).unwrap();
```

Comment:
that's kind of interesting in that the enum is used here but allow dead code is required.

Reply:
Let me double-check.

Reply:
So the linter complains that we never read the value in the field `a` of MyEnum::Struct (which is true).